modules/launcher/src/main/java/net/java/sip/communicator/launcher/Jitsi.java

```diff
@@ -101,17 +95,7 @@ public class Jitsi
      * @param args command line args if any
      * @throws Exception whenever it makes sense.
      */
-    public static void main(String[] args)
```

Comment:
Keep this. I've checked again, and it really is needed, even with the other changes below. Set a breakpoint in any Activator loaded from the "framework", you'll see that it's loaded from AppClassLoader instead of the BundleClassLoader.

Why is that? Contrary to a real OSGi framework, this fake-thing doesn't isolate jars and their classes. Especially not the ones directly specified at the JVM level. Since _all_ of the classes are specified there, all classloading bubbles up (which is specified behavior of ClassLoaders). This breaks native library loading. You mentioned it worked for you, but this is not reliable, e.g. on Windows WASAPI still loads because JNA magically finds it at the right place, but other stuff is missing (jnffmpeg, hid, portaudio, ...).

Reply:
I investigated for myself. See comment.